frontends/android/BitBoxApp/app/src/main/java/ch/shiftcrypto/bitboxapp/MainActivity.java

```diff
@@ -467,6 +493,30 @@ public void run() {
                 });
             }
         });
+
+        connectivityManager = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
+        networkCallback = new ConnectivityManager.NetworkCallback() {
+            @Override
+            public void onCapabilitiesChanged(android.net.Network network, android.net.NetworkCapabilities capabilities) {
+                super.onCapabilitiesChanged(network, capabilities);
+                Mobileserver.setOnline(hasInternetConnectivity(capabilities));
+            }
+            // When we lose the network, onCapabilitiesChanged does not trigger, so we need to override onLost.
+            @Override
+            public void onLost(android.net.Network network) {
+                super.onLost(network);
+                Mobileserver.setOnline(false);
+            }
+            // The same is true for when the network is available again.
+            @Override
+            public void onAvailable(android.net.Network network) {
+                super.onAvailable(network);
+                Mobileserver.setOnline(true);
```

Comment:
onAvailable means there is a link (cable, WiFi), but not necessarily that you are online. E.g. WiFi catptive portals would return true here while not being connected. 

use 

```
NetworkCapabilities capabilities = connectivityManager.getNetworkCapabilities(network);
Mobileserver.setOnline(hasInternetConnectivity(capabilities));
```